app/components/channel_list/categories/unreads.tsx

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+import {FlatList, Text} from 'react-native';
+
+import {useTheme} from '@app/context/theme';
+import {changeOpacity, makeStyleSheetFromTheme} from '@app/utils/theme';
+import {typography} from '@app/utils/typography';
+
+import ChannelListItem from './body/channel';
+
+import type ChannelModel from '@typings/database/models/servers/channel';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    heading: {
+        color: changeOpacity(theme.sidebarText, 0.64),
+        ...typography('Heading', 75),
+        paddingLeft: 5,
+        paddingTop: 10,
+    },
+}));
+
+const UnreadCategories = ({unreadChannels}: {unreadChannels: ChannelModel[]}) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const intl = useIntl();
+
+    const renderItem = ({item}: {item: ChannelModel}) => {
```

Comment:
Should probably be a callback. A re-render on on UnreadCategories may trigger a re-render on the FlatList, but that should be smart enough to just render the changed components. If this is not a callback, we will send a new renderItem function, that will re-render every single element.

Reply:
I did thought about this, but for me it made sense as it only needs to re-render if `unreadChannels` change, but in case the parent component re-renders this will cause the list to re-render as well, blah blah..

What I see that would be even better than a  `useCallback`  is to extract the function from the component as it only needs the item being passed as an argument.